k9mail/src/main/java/com/fsck/k9/controller/SyncHelper.java

```diff
@@ -0,0 +1,262 @@
+package com.fsck.k9.controller;
+
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import android.content.Context;
+
+import com.fsck.k9.Account;
+import com.fsck.k9.K9;
+import com.fsck.k9.cache.EmailProviderCache;
+import com.fsck.k9.helper.Contacts;
+import com.fsck.k9.mail.Flag;
+import com.fsck.k9.mail.Folder;
+import com.fsck.k9.mail.Folder.FolderType;
+import com.fsck.k9.mail.Message;
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mailstore.LocalFolder;
+import com.fsck.k9.mailstore.LocalFolder.MoreMessages;
+import com.fsck.k9.mailstore.LocalMessage;
+import timber.log.Timber;
+
+
+class SyncHelper {
+
+    private static SyncHelper INSTANCE;
```

Comment:
Why do you need an instance at all - these all seem to be static helper methods.

Also, this class could benefit from tests.

One idea is to make it an object, but inject it into classes. Then you could mock it. Only do that if it's useful though. If it's just a utils class then make it just a bunch of static methods and test them individually (as well as just using them in the rest of the code).

Reply:
I seem to remember that these methods used to be static, but then there were some complications in testing. I'll look again.